compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/kotlin/KSValueArgumentImpl.kt

```diff
@@ -26,15 +26,22 @@ import com.google.devtools.ksp.symbol.KSVisitor
 import com.google.devtools.ksp.symbol.Location
 import com.google.devtools.ksp.symbol.NonExistLocation
 import com.google.devtools.ksp.symbol.Origin
+import java.util.Objects
 
 class KSValueArgumentLiteImpl private constructor(
     override val name: KSName,
     override val value: Any?,
-    override val parent: KSNode?
+    override val parent: KSNode?,
+    override val isDefault: Boolean
 ) : KSValueArgumentImpl() {
-    companion object : KSObjectCache<Triple<KSName, Any?, KSNode>, KSValueArgumentLiteImpl>() {
-        fun getCached(name: KSName, value: Any?, parent: KSNode) = cache
-            .getOrPut(Triple(name, value, parent)) { KSValueArgumentLiteImpl(name, value, parent) }
+    companion object : KSObjectCache<Int, KSValueArgumentLiteImpl>() {
+        private fun getKey(name: KSName, value: Any?, parent: KSNode, isDefault: Boolean): Int {
+            return Objects.hash(name, value, parent, isDefault)
+        }
+        fun getCached(name: KSName, value: Any?, parent: KSNode, isDefault: Boolean) = cache
+            .getOrPut(getKey(name, value, parent, isDefault)) {
+                KSValueArgumentLiteImpl(name, value, parent, isDefault)
+            }
     }
 
     override val origin = Origin.KOTLIN
```

Comment:
On a side note, i am thinking whether we should put `Origin.SYNTHETIC` for default annotation arguments, as they are not technically present in the source code, I can do it myself afterwards though, feel free to ignore.